src/core_plugins/input_control_vis/public/control/list_control_factory.js

```diff
@@ -117,6 +117,10 @@ class ListControl extends Control {
     this.enable = true;
     this.disabledReason = '';
   }
+
+  hasValue() {
+    return this.value && this.value.length > 0;
```

Comment:
I think there is a bug here. Looks like I can't set an input control drop down to false.  Should probably be a hard check for undefined.  Could you add a test that would have caught this issue when you fix it?

Reply:
The value can never be set to `false`. For list control, the value is either undefined for an array of objects with the shape label and value. I will create a test to show how the value gets set

Reply:
Ah, okay.  I suppose I am hitting another bug.  I created a list drop down with true and false values for the sample flight data set and I get: 
```
Uncaught (in promise) Error: 0 is not a valid boolean value for boolean field Cancelled
    at getConvertedValueForField (phrase.js:71)
    at buildPhraseFilter (phrase.js:31)
    at PhraseFilterManager.createFilter (phrase_filter_manager.js:72)
    at ListControl.set (control.js:154)
    at VisController._callee3$ (vis_controller.js:190)
    at tryCatch (runtime.js:65)
    at Generator.invoke [as _invoke] (runtime.js:303)
    at Generator.prototype.(anonymous function) [as next] (http://localhost:5601/xmb/bundles/vendors.bundle.js:218445:21)
    at step (vis_controller.js:41)
    at vis_controller.js:41
    at new Promise (<anonymous>)
    at VisController.<anonymous> (vis_controller.js:41)
    at VisController.stageFilter (vis_controller.js:218)
    at ListControl._this.handleOnChange (list_control.js:64)
    at EuiComboBox.onAddOption (combo_box.js:630)
    at onOptionClick (combo_box.js:621)
```

Once we have this typescripted we can enforce that `set` on the base class is only ever called with a certain type, but as it is now, it's not clear that `set` should only be called with an array type or undefined.  But I suppose this is fine as is for this PR.

Reply:
Would you mind creating a new issue for using boolean fields?